utils/network.go

```diff
@@ -6,151 +6,229 @@ import (
+
+	n.blockHashMetaInfo = &blockHashMetaInfo{}
+
+	var err error
+	if n.blockHashMetaInfo.FallBackSequencerAddress, err = new(felt.Felt).SetString(elems[5]); err != nil {
+		return ErrNetworkSetFallbackAddr
+	}
+	if n.blockHashMetaInfo.First07Block, err = strconv.ParseUint(elems[6], 10, 64); err != nil {
+		return ErrNetworkSetFirst07Block
+	}
+	var start, end uint64
+	if start, err = strconv.ParseUint(elems[7], 10, 64); err != nil {
+		return ErrNetworkSetUnverifRangeStart
+	}
+	if end, err = strconv.ParseUint(elems[8], 10, 64); err != nil {
+		return ErrNetworkSetUnverifRangeEnd
+	}
+	n.blockHashMetaInfo.UnverifiableRange = []uint64{start, end}
 	return nil
 }
 
-func (n *Network) Type() string {
+func (n Network) Type() string {
 	return "Network"
 }
 
 func (n *Network) UnmarshalText(text []byte) error {
 	return n.Set(string(text))
 }
 
-// baseURL returns the base URL without endpoint
-func (n Network) baseURL() string {
-	switch n {
-	case GOERLI:
-		return "https://alpha4.starknet.io/"
-	case MAINNET:
-		return "https://alpha-mainnet.starknet.io/"
-	case GOERLI2:
-		return "https://alpha4-2.starknet.io/"
-	case INTEGRATION:
-		return "https://external.integration.starknet.io/"
-	default:
-		// Should not happen.
-		panic(ErrUnknownNetwork)
-	}
-}
-
 // FeederURL returns URL for read commands
 func (n Network) FeederURL() string {
-	return n.baseURL() + "feeder_gateway/"
+	return n.baseURL + "feeder_gateway/"
 }
 
 // GatewayURL returns URL for write commands
 func (n Network) GatewayURL() string {
-	return n.baseURL() + "gateway/"
+	return n.baseURL + "gateway/"
 }
 
 func (n Network) ChainIDString() string {
-	switch n {
-	case GOERLI, INTEGRATION:
-		return "SN_GOERLI"
-	case MAINNET:
-		return "SN_MAIN"
-	case GOERLI2:
-		return "SN_GOERLI2"
-	default:
-		// Should not happen.
-		panic(ErrUnknownNetwork)
-	}
+	return n.chainID
 }
 
 func (n Network) DefaultL1ChainID() *big.Int {
-	var chainID int64
-	switch n {
-	case MAINNET:
-		chainID = 1
-	case GOERLI, GOERLI2, INTEGRATION:
-		chainID = 5
-	default:
-		// Should not happen.
-		panic(ErrUnknownNetwork)
-	}
-	return big.NewInt(chainID)
+	return n.l1ChainID
 }
 
 func (n Network) CoreContractAddress() (common.Address, error) {
-	var address common.Address
-	// The docs states the addresses for each network: https://docs.starknet.io/documentation/useful_info/
-	switch n {
-	case MAINNET:
-		address = common.HexToAddress("0xc662c410C0ECf747543f5bA90660f6ABeBD9C8c4")
-	case GOERLI:
-		address = common.HexToAddress("0xde29d060D45901Fb19ED6C6e959EB22d8626708e")
-	case GOERLI2:
-		address = common.HexToAddress("0xa4eD3aD27c294565cB0DCc993BDdCC75432D498c")
-	case INTEGRATION:
-		return common.Address{}, errors.New("l1 contract is not available on the integration network")
-	default:
-		// Should not happen.
-		return common.Address{}, ErrUnknownNetwork
-	}
-	return address, nil
+	if n.l1ChainID == nil {
+		return common.Address{}, errors.New("l1 contract is not available on this network")
```

Comment:
```suggestion
		return common.Address{}, fmt.Errorf("l1 contract is not available on network %s", n.name)
```
nit: make error message slightly more informative